### PR TITLE
Refactor to index and single example pages

### PR DESCRIPTION
**Pull request by lddelchev**
Ticket: https://kanbanzone.io/b/YA2t2Uox/c/762-Revamp-Usage-Examples-index-view

test page: https://developer.transportapi.com/examplestest

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -47,10 +47,17 @@
             integrity="sha512-WFN04846sdKMIP5LKNphMaWzU7YpMyCU245etK3g/2ARYbPK9Ub18eG+ljU96qKRCWh+quCY7yefSmlkQw1ANQ=="
             crossorigin="anonymous">
     </script>
+
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css">
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (katya-spasova) [inline src/index.html:51]: This will be included only for local development, right? Let's make sure we don't include the `js` file twice in production. As we saw this causes problems.

File: tmp/tapi-dev-portal-2022-old.css [hunk 2]
```diff
@@ -0,0 +1,386 @@
+body {
```
Comment (katya-spasova) [inline tmp/tapi-dev-portal-2022-old.css]: Is this file needed?

Comment (lddelchev) [reply tmp/tapi-dev-portal-2022-old.css]: No, I will remove it.

**Comment by lddelchev** [comment]
I will add 1 more commit to this PR - Refactoring to pre-compile handlebars templates.

**Review by katya-spasova — APPROVED** [review]
LGTM

File: Readme.md [hunk 3]
```diff
@@ -10,31 +10,27 @@ we don't want functional code getting lost in a sea of styling.
...
-The page of usage examples use [CodeSandbox](https://codesandbox.io/)
-for the embedded source playground.
-It is online and to load it with your local changes reflected you need to
-push your development git branch and add its name as request parameter `branch`.
-
-Example:
 
-`http://localhost:8080/src?example=bus-stop-timetable&localDevelopment=true&branch=<YOUR DEVELOPMENT BRANCH>`
+3. Open
+`http://localhost:1234/?localDevelopment=true&branch=<YOUR DEVELOPMENT BRANCH>`
```
Comment (vdimchev) [inline Readme.md:27]: Do we still need the `localDevelopment` and `branch` query string params?

File: src/main.js [hunk 4]
```diff
@@ -1,7 +1,17 @@
-/* global $ _ Prism Handlebars */
+/* global $ _ Prism */
+import filePartial from './partials/file.hbs'
+import ctaSectionPartial from './partials/cta_section.hbs'
+import navigationLayout from './layouts/navigation.hbs'
+import exampleLayout from './layouts/example.hbs'
+
+// production should be set in index.html
+// eslint-disable-next-line no-undef
+const STATIC_CONTENT_URL = production ? 'https://examples.transportapi.com/' : 'https://examples.staging.transportapi.com/'
```
Comment (vdimchev) [inline src/main.js:9]: Good to document somewhere `production`, it's not obvious at all and it's kind of external interface.

File: src/main.js [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
+const RAW_GITHUB_CONTENT_URL = 'https://raw.githubusercontent.com/transportapi/usage-examples/master/src/'
+const EXAMPLES_SOURCE_GITHUB_URL = 'https://github.com/transportapi/usage-examples/tree/master/src/examples/'
+const PRODUCT_PAGES_ROOT = 'https://www.transportapi.com/managed-services/'
+const LEGACY_DOCS_URL = 'https://developer.transportapi.com/docs?raml=https://docs.transportapi.com/raml/transportapi.raml'
```
Comment (vdimchev) [inline src/main.js]: Huh, I guess we should start pointing to the new documentation when an endpoint is migrated. We can do it even nor for the `public_journey`. 

But let's not block the rollout and do it on a second pass.

Comment (vdimchev) [inline src/main.js]: Let's be consistent when naming constants. Either all capitals or camel case as below.

File: src/main.js [hunk 6]
```diff
@@ -12,114 +22,250 @@ const showExperimental =
...
+      height: 400
+    }
   },
   {
     title: 'Journey planner',
     description: 'Shows a journey plan between two points',
     directory: 'journey-planner',
-    experimental: false
+    endpoint: {
+      path: 'public/journey/from/{from}/to/{to}',
```
Comment (vdimchev) [inline src/main.js:180]: I'm noting so we don't forget - migrate the example to the new endpoint and point to the new docs.

Not for this PR.

File: src/main.js [hunk 7]
```diff
@@ -12,114 +22,250 @@ const showExperimental =
...
 
 if (urlParams.has('example')) {
-  const directory = urlParams.get('example')
-  const properties = _.find(examples, { directory })
-  showExample(properties)
+  const exampleName = urlParams.get('example')
+  showExample(exampleName)
 } else {
   const examplesToList = showExperimental ? examples : examples.filter(example => !example.experimental)
+  const productExamples = organizeByProduct(examplesToList)
```
Comment (vdimchev) [inline src/main.js:227]: I start to miss TypeScript here :trollface:

File: src/main.js [hunk 8]
```diff
@@ -12,114 +22,250 @@ const showExperimental =
...
   const properties = {
-    examples: examplesToList,
+    productExamples: productExamples,
     additionalUrlQueryParams: exampleViewUrlQueryParameters(),
     showExample: showExample
   }
-  loadTemplate('navigation.hbs', properties, () => null)
+  loadTemplate(navigationLayout, properties, () => null)
```
Comment (vdimchev) [inline src/main.js:233]: So you actually call the layout here? The actual `exampe.hbs` for example?

Is it some artifact of partials pre-compiling?

File: src/layouts/example.hbs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+    <div class="row">
+        <div class="col-md-12 pt-3">
+            <div class="d-flex flex-row align-items-start">
+                <div class="pe-2">
+                    API Endpoint:
+                </div>
+                <a href="{{endpoint.docsUrl}}" class="crosslink ps-1">
+                    {{endpoint.path}}
+                </a>
```
Comment (vdimchev) [inline src/layouts/example.hbs:38]: Maybe open the docs in another tab (`target=_blank`)?

File: src/layouts/example.hbs [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+                <div class="pe-2">
+                    Product:
+                </div>
+                <a href="{{product.page}}" class="crosslink text-center pe-1" style="width:3rem">
+                    <img class="img-fluid px-1" src="{{rootUrl}}{{product.logo}}">
+                </a>
+                <a href="{{product.page}}" class="crosslink">
+                    {{product.label}}
+                </a>
+            </div>
```
Comment (vdimchev) [inline src/layouts/example.hbs:51]: Same as above, open in another tab?

**Review by vdimchev — APPROVED** [review]
Some things for future improvements, but otherwise - awesome!
